flink-libraries/flink-table/src/test/scala/org/apache/flink/api/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -1101,6 +1101,45 @@ class ScalarFunctionsTest extends ExpressionTestBase {
       "true")
   }
 
+  @Test
+  def testInExpressions(): Unit = {
+    testTableApi(
```

Comment:
Yes, we can also create a follow-up issue for that. But since we are not only building a SQL API but also a Table API, POJOs will occur very frequently and everywhere. We should support them as good as we can.